R/deprecated.R

```diff
@@ -102,14 +102,15 @@ deprecate_soft <- function(when,
   verbosity <- lifecycle_verbosity()
   if (verbosity == "quiet") {
     NULL
-  } else if (verbosity %in% "warning" ||
-             (is_string(verbosity, "default") && env_inherits_global(user_env))) {
-    trace <- trace_back(bottom = caller_env())
-    deprecate_warn0(msg, trace, always = TRUE)
+  } else if (verbosity %in% c("warning", "default")) {
+    if (env_inherits_global(user_env) || from_testthat(user_env)) {
+      trace <- trace_back(bottom = caller_env())
+      deprecate_warn0(msg, trace, always = verbosity == "warning")
+    } else {
+      deprecate_soft0(msg)
```

Comment:
Inline `deprecate_soft0()` or rename to `signal_soft_deprecation()`? The name confused me for a while.

Reply:
I refactored both `deprecate_soft()` and `deprecate_warn()` to make the logic more clear.